diff --git a/configs/transpeed-8k618-t_defconfig b/configs/transpeed-8k618-t_defconfig
index 221614762e8..11ddd535064 100644
--- a/configs/transpeed-8k618-t_defconfig
+++ b/configs/transpeed-8k618-t_defconfig
@@ -1,6 +1,7 @@
 CONFIG_ARM=y
 CONFIG_ARCH_SUNXI=y
 CONFIG_DEFAULT_DEVICE_TREE="allwinner/sun50i-h618-transpeed-8k618-t"
+CONFIG_DRAM_CLK=648
 CONFIG_SPL=y
 CONFIG_DRAM_SUNXI_DX_ODT=0x03030303
 CONFIG_DRAM_SUNXI_DX_DRI=0x0e0e0e0e
@@ -11,7 +12,6 @@ CONFIG_DRAM_SUNXI_TPR11=0xddddcccc
 CONFIG_DRAM_SUNXI_TPR12=0xeddc7665
 CONFIG_MACH_SUN50I_H616=y
 CONFIG_SUNXI_DRAM_H616_DDR3_1333=y
-CONFIG_DRAM_CLK=648
 CONFIG_MMC_SUNXI_SLOT_EXTRA=2
 CONFIG_R_I2C_ENABLE=y
 # CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
@@ -19,9 +19,15 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_SYS_I2C_LEGACY=y
 CONFIG_SYS_I2C_MVTWSI=y
 CONFIG_SYS_I2C_SLAVE=0x7f
-CONFIG_SYS_I2C_SPEED=400000
 CONFIG_SUPPORT_EMMC_BOOT=y
 CONFIG_AXP313_POWER=y
-CONFIG_AXP_DCDC3_VOLT=1360
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_OHCI_HCD=y
+# required for internal ethernet phy most probably
+CONFIG_AXP_DCDC3_VOLT=1500
+CONFIG_SYS_I2C_SPEED=100000
+CONFIG_SPL_I2C_SUPPORT=y
+CONFIG_PHY_REALTEK=y
+CONFIG_SUN8I_EMAC=y
+CONFIG_I2C3_ENABLE=y
+CONFIG_USB_MUSB_GADGET=y
diff --git a/dts/upstream/src/arm64/allwinner/sun50i-h618-transpeed-8k618-t.dts b/dts/upstream/src/arm64/allwinner/sun50i-h618-transpeed-8k618-t.dts
index f828ca1ce51..cf56f4cd296 100644
--- a/dts/upstream/src/arm64/allwinner/sun50i-h618-transpeed-8k618-t.dts
+++ b/dts/upstream/src/arm64/allwinner/sun50i-h618-transpeed-8k618-t.dts
@@ -73,9 +73,32 @@
 	status = "okay";
 };
 
+&i2c3 {
+       status = "okay";
+};
+
+&emac1 {
+       pinctrl-names = "default";
+       pinctrl-0 = <&rmii_pins>;
+       phy-mode = "rmii";
+       phy-handle = <&rmii_phy>;
+       phy-supply = <&reg_aldo1>;
+       allwinner,rx-delay-ps = <3100>;
+       allwinner,tx-delay-ps = <700>;
+       status = "okay";
+};
+
+&mdio1 {
+       rmii_phy: ethernet-phy@16 {
+               compatible = "ethernet-phy-ieee802.3-c22";
+               reg = <16>;
+       };
+};
+
 &mmc0 {
 	vmmc-supply = <&reg_dldo1>;
-	cd-gpios = <&pio 8 16 GPIO_ACTIVE_LOW>;	/* PI16 */
+//	cd-gpios = <&pio 8 16 GPIO_ACTIVE_LOW>;	/* PI16 */
+	broken-cd;
 	disable-wp;
 	bus-width = <4>;
 	status = "okay";
@@ -87,6 +110,7 @@
 	mmc-pwrseq = <&wifi_pwrseq>;
 	bus-width = <4>;
 	non-removable;
+	max-frequency = <25000000>;
 	status = "okay";
 
 	sdio_wifi: wifi@1 {
