diff --git a/arch/arm/dts/rk3328-u-boot.dtsi b/arch/arm/dts/rk3328-u-boot.dtsi
index 1633558264..f5ea6f14df 100644
--- a/arch/arm/dts/rk3328-u-boot.dtsi
+++ b/arch/arm/dts/rk3328-u-boot.dtsi
@@ -5,13 +5,16 @@
 
 / {
 	aliases {
-		mmc0 = &emmc;
-		mmc1 = &sdmmc;
+//		mmc0 = &emmc;
+//		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &emmc;
 		spi0 = &spi0;
 	};
 
 	chosen {
-		u-boot,spl-boot-order = &emmc, &sdmmc;
+//		u-boot,spl-boot-order = &emmc, &sdmmc;
+		u-boot,spl-boot-order = &sdmmc, &emmc;
 	};
 
 	dmc: dmc {
