diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 60b13655f8..ed707cdb95 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -607,5 +607,6 @@ dtb-$(CONFIG_MACH_SUN50I_H6) += \
 	sun50i-h6-pine-h64.dtb
 dtb-$(CONFIG_MACH_SUN50I_H616) += \
-	sun50i-h616-t95.dtb \
+	sun50i-h313-x96q.dtb \
+	sun50i-h616-tx6s.dtb \
 	sun50i-h616-orangepi-zero2.dtb
 dtb-$(CONFIG_MACH_SUN50I) += \
 	sun50i-a64-amarula-relic.dtb \
