diff --git a/configs/odroid_defconfig b/configs/odroid_defconfig
index 345cc3b940..6a322c2832 100644
--- a/configs/odroid_defconfig
+++ b/configs/odroid_defconfig
@@ -65,3 +65,5 @@ CONFIG_USB_HOST_ETHER=y
 CONFIG_USB_ETHER_SMSC95XX=y
 CONFIG_LIB_HW_RAND=y
 CONFIG_ERRNO_STR=y
+CONFIG_BOOTDELAY=5
+CONFIG_USE_PREBOOT=y
diff --git a/include/configs/odroid.h b/include/configs/odroid.h
index 03229910b2..03bdee604f 100644
--- a/include/configs/odroid.h
+++ b/include/configs/odroid.h
@@ -166,6 +166,15 @@
 	"scriptaddr=0x42000000\0" \
 	"fdt_addr_r=0x40800000\0" \
 	"kernel_addr_r=0x41000000\0" \
+	"bootenv_addr_r=0x40800000\0" \
+	"preboot=for target in 1 2;do " \
+		"for prefix in / /boot/;do " \
+			"if test -e mmc ${target}:1 ${prefix}uEnv.txt;then " \
+				"load mmc ${target}:1 ${bootenv_addr_r} ${prefix}uEnv.txt; " \
+				"env import -t ${bootenv_addr_r} ${filesize}; " \
+			"fi; " \
+		"done; " \
+	"done\0" \
 	BOOTENV
 
 /* GPT */
