diff --git a/configs/libretech-cc_defconfig b/configs/libretech-cc_defconfig
index 221f5d52b49..1f33c3911c5 100644
--- a/configs/libretech-cc_defconfig
+++ b/configs/libretech-cc_defconfig
@@ -13,7 +13,7 @@ CONFIG_MESON_GXL=y
 CONFIG_SYS_LOAD_ADDR=0x1000000
 CONFIG_DEBUG_UART_BASE=0xc81004c0
 CONFIG_DEBUG_UART_CLOCK=24000000
-CONFIG_IDENT_STRING=" libretech-cc"
+CONFIG_IDENT_STRING=" hexdump-gxl"
 CONFIG_DEBUG_UART=y
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
@@ -79,3 +79,8 @@ CONFIG_VIDEO_BMP_RLE8=y
 CONFIG_BMP_16BPP=y
 CONFIG_BMP_24BPP=y
 CONFIG_BMP_32BPP=y
+CONFIG_DISPLAY_BOARDINFO_LATE=y
+CONFIG_BOOTDELAY=5
+CONFIG_USE_PREBOOT=y
+CONFIG_LED=y
+CONFIG_LED_GPIO=y
diff --git a/dts/upstream/src/arm64/amlogic/meson-gxl-s905x-libretech-cc.dts b/dts/upstream/src/arm64/amlogic/meson-gxl-s905x-libretech-cc.dts
index 401064b0428..cc409116f34 100644
--- a/dts/upstream/src/arm64/amlogic/meson-gxl-s905x-libretech-cc.dts
+++ b/dts/upstream/src/arm64/amlogic/meson-gxl-s905x-libretech-cc.dts
@@ -15,7 +15,7 @@
 / {
 	compatible = "libretech,aml-s905x-cc", "amlogic,s905x",
 		     "amlogic,meson-gxl";
-	model = "Libre Computer AML-S905X-CC";
+	model = "hexdump usbkbd/hdmi u-boot gxl";
 
 	aliases {
 		serial0 = &uart_AO;
@@ -63,17 +63,18 @@
 		compatible = "gpio-leds";
 
 		led-system {
-			label = "librecomputer:system-status";
+//			label = "librecomputer:system-status";
+			label = "system-status";
 			gpios = <&gpio GPIODV_24 GPIO_ACTIVE_HIGH>;
 			default-state = "on";
 			panic-indicator;
 		};
 
-		led-blue {
-			label = "librecomputer:blue";
-			gpios = <&gpio_ao GPIOAO_2 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
-		};
+// 		led-blue {
+// 			label = "librecomputer:blue";
+// 			gpios = <&gpio_ao GPIOAO_2 GPIO_ACTIVE_HIGH>;
+// 			linux,default-trigger = "heartbeat";
+// 		};
 	};
 
 	memory@0 {
diff --git a/include/configs/meson64.h b/include/configs/meson64.h
index f3275b37a51..f41fddeae6e 100644
--- a/include/configs/meson64.h
+++ b/include/configs/meson64.h
@@ -145,6 +145,8 @@
 #endif
 
 #ifndef CFG_EXTRA_ENV_SETTINGS
+/* the led system-status on is a hack to turn on the led as for some strange reason it does not
+ * the usb start below makes sure a usb keyboard is available when the autoboot prompt starts */
 #define CFG_EXTRA_ENV_SETTINGS \
 	"stdin=" STDIN_CFG "\0" \
 	"stdout=" STDOUT_CFG "\0" \
@@ -159,6 +161,17 @@
 	"ramdisk_addr_r=" RAMDISK_ADDR_R "\0" \
 	"fdtfile=" FDTFILE_NAME "\0" \
 	"dfu_alt_info=fitimage ram " KERNEL_ADDR_R " 0x4000000 \0" \
+	"bootenv_addr_r=0x01080000\0" \
+	"preboot=led system-status on; " \
+		"usb start; " \
+		"for target in 0 1;do " \
+		"for prefix in / /boot/;do " \
+			"if test -e mmc ${target}:1 ${prefix}uEnv.txt;then " \
+				"load mmc ${target}:1 ${bootenv_addr_r} ${prefix}uEnv.txt; " \
+				"env import -t ${bootenv_addr_r} ${filesize}; " \
+			"fi; " \
+		"done; " \
+	"done\0" \
 	BOOTENV
 #endif
 
